core/src/graph/mod.rs

```diff
@@ -186,6 +189,106 @@ derive_from! {
     }
 }
 
+impl Display for Value {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", serde_json::to_string(&self).unwrap())
+    }
+}
+
+#[allow(unused)]
+impl Value {
+    pub fn as_null(&self) -> Option<()> {
+        match *self {
+            Value::Null(()) => Some(()),
+            _ => None
+        }
+    }
+
+    pub fn as_bool(&self) -> Option<&bool> {
+        match *self {
+            Value::Bool(ref bool) => Some(bool),
```

Comment:
I don't like that this shadows the `bool` type. Also the whole thing looks like those impls be shortened considerably by using a macro.

Reply:
This is how `serde_json` implements these `as_x` functions as well.